### PR TITLE
Builder documentation: Pagination documentation

### PR DESCRIPTION
**Pull request by lmossman**
## What
Related to https://github.com/airbytehq/airbyte/issues/24949

(not complete, screen recording still missing)

### CODE REVIEW

**Review by flash1293 — COMMENTED** [review]
Reads pretty well, I had some smaller remarks, feel free to dismiss them as most of them are pretty subjective.

File: docs/connector-development/connector-builder-ui/pagination.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
+# Pagination
+
+Pagination is a mechanism used by APIs in which data is split up into "pages" when returning results, so that the entire response data doesn't need to be returned all at once.
+
+The Connector Builder offers a Pagination section which implements the most common pagination methods used by APIs. When enabled, the connector will use the pagination configuration you have provided to request consecutive pages of data from the API until there are no more pages to fetch.
+
+If your API doesn't support pagination, simply leave the Pagination section disabled.
```
[the remark this answers is not in the document]
Comment (girarda) [reply docs/connector-development/connector-builder-ui/pagination.md:7]: my recommendation is as big as the API supports unless there's a reason to fetch smaller pages

File: docs/connector-development/connector-builder-ui/pagination.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+Check the documentation of the API you want to integrate to find which type of pagination is uses. Many API docs have a "Pagination" or "Paging" section that describes this.
+
+The following pagination mechanisms are supported in the connector builder:
+
+- [Offset Increment](#offset-increment)
+- [Page Increment](#page-increment)
+- [Cursor Pagination](#cursor-pagination)
+
+Select the matching pagination method for your API and check the sections below for more information about individual methods.
```
Comment (girarda) [inline docs/connector-development/connector-builder-ui/pagination.md:19]: nit: should we tell them to use low-code or the Python CDK if need another type of pagination? We can also link the reader to their respective docs

File: docs/connector-development/connector-builder-ui/pagination.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+  {"id": 1, "name": "Product A"},
+  {"id": 2, "name": "Product B"},
+  {"id": 3, "name": "Product C"},
+  {"id": 4, "name": "Product D"},
+  {"id": 5, "name": "Product E"},
+  {"id": 6, "name": "Product F"}
+]
+```
+
+Then the API may take in a request like this: `GET https://api.example.com/products?limit=2&offset=3`, which could result in the following response:
```
[the remark this answers is not in the document]
Comment (lmossman) [reply docs/connector-development/connector-builder-ui/pagination.md:43]: Great suggestion, added!

**Review by girarda — APPROVED** [review]

File: docs/connector-development/connector-builder-ui/pagination.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (lmossman) [reply docs/connector-development/connector-builder-ui/pagination.md:7]: Added a note about that 👍

File: docs/connector-development/connector-builder-ui/pagination.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+### Page Increment
+
+If your API paginates using page increments, the API docs will likely contain one of the following keywords:
+
+- `page size` / `page_size` / `pagesize` / `per_page`
+- `page number` / `page_number` / `pagenum` / `page`
+
+In this method of pagination, the "page size" specifies the maximum number of records to return per request, while the "page number" indicates the specific page of data to retrieve.
+
+This is similar to Offset Increment pagination, but the main difference lies in how the starting point of the data subset being returned by the request is determined. In Offset Increment pagination, the starting point is explicitly defined by the offset value; in Page Increment pagination, the starting point can be determined by calculating `pageSize * pageNum` if the starting page is `0`, or `pageSize * (pageNum - 1)` if the starting page is `1`.
```
[the remark this answers is not in the document]
Comment (lmossman) [reply docs/connector-development/connector-builder-ui/pagination.md]: I was going back and forth myself on if it made sense to include these details or not, and after reading your comment I agree that it is unnecessary, and your suggestion feels easier to understand 👍

File: docs/connector-development/connector-builder-ui/pagination.md (hunk 2 above)
Comment (lmossman) [reply docs/connector-development/connector-builder-ui/pagination.md]: done 👍

**Review by flash1293 — APPROVED** [review]
LGTM, thanks for these changes

**Pull request merged**